core_impl/src/main/java/io/opencensus/stats/MutableAggregation.java

```diff
@@ -33,6 +39,20 @@ private MutableAggregation() {
   abstract void add(double value);
 
   /**
+   * Update the summary stats by removing an expired value.
+   *
+   * @param value the expired value to be removed from population.
+   */
+  abstract void remove(double value);
```

Comment:
Another idea is to keep the approach used in this PR for sum, count, and histogram, but remove support for stddev, mean, and range.  The current approach should work well for the three exposed aggregations.  When we add the other three, we can decide how to optimize interval views, and maybe start breaking the interval into sub intervals to avoid storing all values.